Added some more compatiblity information

snoack opened the pull request:
These are the information I can add with high confidence from the top of my head.

Note: I didn't touch compatibility information of AdBlock, since AdBlock for Chrome, Firefox and Microsoft Edge are separate implementations. AdBlock for Chrome is based on Adblock Plus, and the features map one-to-one, AdBlock for Firefox and Microsoft Edge, however, are two separate forks of the original AdBlock code base. Any suggestion how to deal with that?

snoack committed:
Added some more compatiblity information

ThomasGreiner commented on data/blocking/_self.json:
```diff
@@ -15,7 +15,11 @@
           },
           "ublockorigin": {
-            "version_added": true
+            "partial_implementation": true,
```
Technically, uBlock Origin does support exception rules which is what this entry represents (i.e. "Basic support" for exception rules). Note that the issue you're pointing out here is mentioned under `blocking.options.document`.

But generally you're right that it is not clearly defined what to understand under "Basic support" so we may need to make up the rules as we go.

snoack replied:
Yeah, I've seen the note under `blocking.options.document`. But (in the current visualization at least) it seemed weird to me that it said "Exception rules: yes" for uBO despite the lack of something as fundamental as document-based whitelisting. Anyway, I undid the change here as requested.

ThomasGreiner reviewed (changes requested):
Thanks for helping out!

> AdBlock for Chrome is based on Adblock Plus, and the features map one-to-one, AdBlock for Firefox and Microsoft Edge, however, are two separate forks of the original AdBlock code base. Any suggestion how to deal with that?

That's indeed quite tricky. What I did with Adblock Plus is reference two version numbers for the time when we released it for Chrome and Firefox independently (e.g. `3.0 (1.13.3)`) since they were both using the same engine (i.e. adblockpluscore). In the case of AdBlock this is not the case so I guess we'd need to add a separate identifier for those (e.g. "adblock_betafish") or find some other solution.

I'd be curious how MDN handles such cases (e.g. back when Opera switched from Presto to Chromium) but I haven't found a similar case there yet.

snoack commented:
> What I did with Adblock Plus is reference two version numbers for the time when we released it for Chrome and Firefox independently (e.g. 3.0 (1.13.3))

Yeah, that was smart. However, if we added a feature in either Adblock Plus for Firefox/Chrome back then, we added it for the other one short after (and meanwhile both are released simultaneously based on the same code under the same version). With different implementations of AdBlock, however, it's possible that a feature is implemented in any number (0-3) of the three ports.

> In the case of AdBlock this is not the case so I guess we'd need to add a separate identifier for those (e.g. "adblock_betafish") or find some other solution.

Technically, even AdBlock for Firefox and AdBlock for Microsoft Edge are two completely different code bases (except for some ancient code inherited from the original AdBlock).

> I'd be curious how MDN handles such cases (e.g. back when Opera switched from Presto to Chromium)

With Opera, it's simple, up to version 14 it was Presto, starting from version 15 it was Chromium. If a feature existed in both, then the `version_added` would be <=14 (or `true`). If it only exists in Chromium, the `version_added` would be >= 15 (or `true`). If it only existed in Presto, the `version_removed` would be 15.

ThomasGreiner reviewed (approved):
LGTM

Regarding AdBlock: Given everything you're saying, it sounds indeed quite tricky so maybe we'll need to oversimplify it a bit, at least for now, for the sake of keeping the data easy to digest.

Anyway, I'll try to dig deeper into their code base to see whether I can find a suitable approach to tackle this issue.
